File: app/src/main/java/com/borisruzanov/russianwives/mvp/presenter/MainPresenter.java
package com.borisruzanov.russianwives.mvp.presenter;

import android.util.Log;

import com.arellomobile.mvp.InjectViewState;
import com.arellomobile.mvp.MvpPresenter;
import com.borisruzanov.russianwives.mvp.model.interactor.MainInteractor;
import com.borisruzanov.russianwives.mvp.view.MainView;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

@InjectViewState
public class MainPresenter extends MvpPresenter<MainView> {

    private MainInteractor mainInteractor;

    public MainPresenter(MainInteractor mainInteractor) {
        this.mainInteractor = mainInteractor;
    }

    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
        getViewState().setViewPager();
    }

    public void checkForUserExist() {
        /*if (mainInteractor.isUserExist()){
            getViewState().setViewPager();
        } else getViewState().callAuthWindow();*/
    }

    public void checkUserRegistered(){
        boolean checkRegistered = mainInteractor.checkUserRegistred();
        Log.d("UzverDebug", "User registered is " + checkRegistered);
    }


    public void saveUser(){
        mainInteractor.saveUser();
    }

    public void userNeedToAddInfo(){
        if (mainInteractor.checkForUserInfo()){
            getViewState().checkingForUserInfo();
        }
    }
}

File: app/src/main/java/com/borisruzanov/russianwives/mvp/ui/main/MainPresenter.kt
package com.borisruzanov.russianwives.mvp.ui.main

import android.util.Log
import com.arellomobile.mvp.InjectViewState
import com.arellomobile.mvp.MvpPresenter
import com.borisruzanov.russianwives.mvp.model.interactor.main.MainInteractor
import com.borisruzanov.russianwives.utils.BoolCallback
import com.borisruzanov.russianwives.utils.StringsCallback
import javax.inject.Inject

@InjectViewState
class MainPresenter @Inject constructor(private val mainInteractor: MainInteractor) : MvpPresenter<MainView>() {

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()

        //mainInteractor.setUserInHot()
//        mainInteractor.getHotUsers(callback = HotUsersCallback { hotUsers ->
//            if (hotUsers.isEmpty()) Log.d("HotUsersDebug", "users is empty")
//            hotUsers.forEach {
//                Log.d("HotUsersDebug", "Uid is ${it.uid}")
//            }
//        })
    }


    fun showGenderFirstDialog(){
        showGenderDialog()
        mainInteractor.setFirstOpenDate()
    }
    fun showDialogs() {
        if (isUserExist()) {
            showMustInfoDialog()
            showAdditionalInfoDialog()
        } else{
            Log.d("AchDebug", "User can write msgs")
        }
    }

    fun checkForUserExist() = viewState.setAdapter(mainInteractor.isUserExist)

    fun isUserExist(): Boolean = mainInteractor.isUserExist

    fun saveUser() = mainInteractor.saveUser()

    fun setGender(gender: String) {
        mainInteractor.setGender(gender)
    }

    fun makeDialogOpenDateDefault() = mainInteractor.makeDialogOpenDateDefault()

    fun openSliderWithDefaults() {
        mainInteractor.getDefaultList(callback = StringsCallback { strings ->
            viewState.openSlider(ArrayList<String>(strings)) })
    }

    /*private fun checkAchieve () {
        RatingRepository().isAchievementExist(FULL_PROFILE_ACH, callback = BoolCallback {
            if (it) Log.d("AchDebug", "User can write msgs")
            else Log.d("AchDebug", "USER CAN'T WRITE")
        })
    }*/

    private fun showGenderDialog() {
        if (mainInteractor.isGenderDefault()) {
//            mView.showGenderDialog()
        }
    }

    fun showMustInfoDialog() {
        mainInteractor.hasDefaultMustInfo(callback = BoolCallback {flag ->
            if (flag) viewState.showMustInfoDialog()
        })
    }

    private fun showAdditionalInfoDialog() {
        mainInteractor.hasNecessaryInfo(callback = BoolCallback { flag ->
            if (flag) {
            viewState.showFullInfoDialog()
            mainInteractor.setFPDialogLastOpenDate()
        }
        })
    }

}
